service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/kubernetesNamespace/DeleteNamespaceRoleStep.java

```diff
@@ -30,15 +30,22 @@ private String getDeletePodName() {
   }
 
   @Override
-  public StepResult doStep(FlightContext context) throws InterruptedException {
+  public StepResult deleteResource(FlightContext context) throws InterruptedException {
     logger.info("Deleting namespace role for namespace {}", resource.getKubernetesNamespace());
-    azureDatabaseUtilsRunner.deleteNamespaceRole(
-        context
-            .getWorkingMap()
-            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class),
-        workspaceId,
-        getDeletePodName(),
-        resource.getKubernetesServiceAccount());
+    try {
+      azureDatabaseUtilsRunner.deleteNamespaceRole(
+          context
+              .getWorkingMap()
+              .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class),
+          workspaceId,
+          getDeletePodName(),
+          resource.getKubernetesServiceAccount());
+    } catch (IllegalStateException e) {
+      // no shared database means no role to delete
+      if (e.getMessage().contains("No shared database found")) {
```

Comment:
This isn't part of your PR but: string matching on the contents of an exception's message to manage control flow is suboptimal. What if we introduce a more targeted exception type to make this less fragile?